addons/base_hash/models/hash_mixin.py

```diff
@@ -0,0 +1,208 @@
+        - The second value is the sorting key
+        - The third value is the id of the record
+            (whenever two records have the same sorting key, we want the one with the smallest id to appear first)
+        """
+        return self.sorted(key=lambda r: (r[r._get_sorting_key()] is False,
+                                          r[r._get_sorting_key()],
+                                          r['id']))
+
+    @api.model
+    def _get_sorting_key(self):
+        """
+        This method must be overriden by the inheriting class.
+        It should return the key on which the records will be sorted.
+        E.g.: 'invoice_date'
+        """
+        raise NotImplementedError("'_get_sort_keys' must be overriden by the inheriting class")
+
+    @api.depends('secure_sequence_number')
+    def _compute_inalterable_hash(self):
+        """
+        Computes the hash of the record using the hash if necessary.
+        See _create_hash_string for more details
+        """
+        for record in self.sorted("secure_sequence_number"):
+            if not record.inalterable_hash and record.must_hash:
+                record.inalterable_hash = record._create_hash_string()
+
+    def _create_hash_string(self, previous_hash=None):
+        """
+        Returns the hash as a string computed with the previous hash
+        and the fields in self._get_fields_used_by_hash
+        """
+        self.ensure_one()
+        # Make the json serialization canonical https://tools.ietf.org/html/draft-staykov-hu-json-canonical-form-00)
+        hash_string = dumps(
+            self._get_dict_fields_values_to_hash(),
+            sort_keys=True, ensure_ascii=True,
+            indent=None, separators=(',', ':')
+        )
+        previous_hash = previous_hash or self._get_previous_hash()
+        hash_string = sha256((previous_hash + hash_string).encode('utf-8'))
+        return hash_string.hexdigest()
+
+    def _get_previous_hash(self):
+        """
+        Returns the hash of the previous record in the secure_sequence chain, or '' if it is the first record.
+        """
+        previous_record = self._get_previous_record_in_secure_sequence()
+        if len(previous_record) > 1:
+            raise UserError(_('An error occurred when computing the inalterability. '
+                              'Impossible to get the unique previous record in the secure chain.'))
+        return previous_record.inalterable_hash if previous_record else ""
+
+    def _get_previous_record_in_secure_sequence(self):
+        """
+        This method must be overriden by the inheriting class.
+        It should return the hash of the previous record in the secure_sequence chain.
+        This previous hash is going to be used to compute the new hash of the current record
+        to create an inalterable hash chain.
+        """
+        raise NotImplementedError("'_get_previous_record_in_secure_sequence' must be overriden by the inheriting class")
```

Comment:
Unfortunately no because of special cases such as this one where the previous record is not necessary the last `secure_sequence_number`
https://github.com/odoo/odoo/pull/84830/files#diff-e048e2746fa92d76e5b2457858422ce24394f292a6d77c2ef78b29ee2203ecc1R28-R37